test/properties/second_quantization/electronic/integrals/test_two_body_electronic_integrals.py

```diff
@@ -0,0 +1,187 @@
+
+        with self.subTest("Missing alpha"):
+            with self.assertRaises(TypeError):
+                TwoBodyElectronicIntegrals(ElectronicBasis.MO, (None, random, random, random))
+
+    def test_transform_basis(self):
+        """Test transform_basis"""
+        mat_aa = np.arange(16).reshape((2, 2, 2, 2))
+        mat_ba = np.arange(16, 32).reshape((2, 2, 2, 2))
+        mat_bb = np.arange(-16, 0).reshape((2, 2, 2, 2))
+
+        transform = ElectronicBasisTransform(ElectronicBasis.AO, ElectronicBasis.MO, 2 * np.eye(2))
+
+        with self.subTest("Pure Alpha"):
+            ints_ao = TwoBodyElectronicIntegrals(ElectronicBasis.AO, (mat_aa, None, None, None))
+            ints_mo = ints_ao.transform_basis(transform)
+            assert np.allclose(ints_mo._matrices[0], 16 * mat_aa)
+            assert ints_mo._matrices[1] is None
+            assert ints_mo._matrices[2] is None
+            assert ints_mo._matrices[3] is None
+
+        with self.subTest("Alpha and Beta"):
+            ints_ao = TwoBodyElectronicIntegrals(
+                ElectronicBasis.AO, (mat_aa, mat_ba, mat_bb, mat_ba.T)
+            )
+            ints_mo = ints_ao.transform_basis(transform)
+            assert np.allclose(ints_mo._matrices[0], 16 * mat_aa)
+            assert np.allclose(ints_mo._matrices[1], 16 * mat_ba)
+            assert np.allclose(ints_mo._matrices[2], 16 * mat_bb)
+            assert np.allclose(ints_mo._matrices[3], 16 * mat_ba.T)
+
+        with self.subTest("Beta custom coeff with only alpha"):
+            transform_beta = ElectronicBasisTransform(
+                ElectronicBasis.AO, ElectronicBasis.MO, 2 * np.eye(2), 3 * np.eye(2)
+            )
+            ints_ao = TwoBodyElectronicIntegrals(ElectronicBasis.AO, (mat_aa, None, None, None))
+            ints_mo = ints_ao.transform_basis(transform_beta)
+            assert np.allclose(ints_mo._matrices[0], 16 * mat_aa)
+            assert ints_mo._matrices[1] is None
+            assert ints_mo._matrices[2] is None
+            assert ints_mo._matrices[3] is None
+
+        with self.subTest("Beta custom coeff"):
+            transform_beta = ElectronicBasisTransform(
+                ElectronicBasis.AO, ElectronicBasis.MO, 2 * np.eye(2), 3 * np.eye(2)
+            )
+            ints_ao = TwoBodyElectronicIntegrals(
+                ElectronicBasis.AO, (mat_aa, mat_ba, mat_bb, mat_ba.T)
+            )
+            ints_mo = ints_ao.transform_basis(transform_beta)
+            assert np.allclose(ints_mo._matrices[0], 16 * mat_aa)
+            assert np.allclose(ints_mo._matrices[1], 36 * mat_ba)
+            assert np.allclose(ints_mo._matrices[2], 81 * mat_bb)
+            assert np.allclose(ints_mo._matrices[3], 36 * mat_ba.T)
+
+        with self.subTest("Final basis match"):
+            ints_ao = TwoBodyElectronicIntegrals(ElectronicBasis.MO, (mat_aa, None, None, None))
+            ints_mo = ints_ao.transform_basis(transform)
+            assert ints_ao == ints_mo
```

Comment:
```suggestion
            self.assertEqual(ints_ao, ints_mo)
```
Applies to other places.